### DISCUSSION TITLE
Can a widget be clickable/draggable without Sense.focusable being true?

### DISCUSSION THREAD

**Discussion by shiver** [emilk/egui#691] [Q&A]
Hi there :wave:

I am making a _very_ custom UI with egui at the moment and it looks like click and drag events will not fire unless `Sense.focusable = true`. However, I explicitly don't want these widgets to take focus as it seems to prevent `hover` events on other widgets from firing?
```
let clickable = ui.allocate_rect(
    rect,
    Sense {
        click: true,
        focusable: false, // Change this to true and the clicked() below works
        drag: false,
    },
);

if clickable.clicked() {
    // This will never fire
}
```

Am I right in saying the `focusable` is required for this? If not, any idea what I might be getting wrong? Any extra information I can provide?

Thanks!

EDIT: So I managed to address my issue with being able to detect when a widget is being hovered by instead using `ui.rect_contains_pointer(hoverable_rect)`, which means that I can safely give the `focusable` sense back to the clickable areas.

However, my question still remains cause I think that being able to use `.hovered()` directly would be a slightly better API. Also, being able to explicitly avoid widgets becoming focus targets seems beneficial to me too.

**Comment by emilk** [comment]
`focusable` means you can focus the widget with the keyboard (pressing tab to go to it).

> I explicitly don't want these widgets to take focus as it seems to prevent hover events on other widgets from firing

I don't know what you mean by this. All interactive widgets have `focusable: true` and it does not prevent any hover events for other widgets.

> Am I right in saying the `focusable` is required for this?

Yes, currently egui assumes that if you don't want to be able to focus a widget with the keyboard, it is because it is non-interactive. Having clickable things that people can't reach with the keyboard seems like a potential confusing UX, and quite bad for accessibility reasons, but there is no technical reason egui couldn't support it.

**Comment by stevenwdv** [comment]
I have a password field with show password feature like in https://github.com/emilk/egui/blob/master/crates/egui_demo_lib/src/demo/password.rs, but I don't want the show password button to be selected when pressing tab, which I think makes sense when filling out forms, so I think this would be a valid usecase for this. Although I don't think existing widgets can have their `Sense` changed.
